src/dawn/Unittest/PrintAllExpressionTypes.cpp

```diff
@@ -0,0 +1,111 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Unittest/PrintAllExpressionTypes.h"
+#include "dawn/SIR/ASTStringifier.h"
+#include "dawn/SIR/SIR.h"
+#include <iostream>
+
+namespace dawn {
+
+void PrintAllExpressionTypes::visit(const std::shared_ptr<BlockStmt>& node) {
+  std::cout << "Block Statement\n" << ASTStringifer::toString(node) << std::endl;
+  for(const auto& s : node->getChildren())
```

Comment:
Should always call the base `visit` function i.e `ASTVisitorForwarding::visit(node);`